qiskit_experiments/framework/base_experiment.py

```diff
@@ -171,13 +179,20 @@ def config(self) -> ExperimentConfig:
             (key, getattr(self._transpile_options, key)) for key in self._set_transpile_options
         )
         run_options = dict((key, getattr(self._run_options, key)) for key in self._set_run_options)
+
```

Comment:
There is an issue with config handling when it comes to serializing a backend object when one is provided in the `__init__` method. In this case, the backend is included in the config e.g. via `kwargs` and is serialized in way that upon deserialization (through `BaseExperiment.load`) the following warning is emitted: 

> /tmp/qiskit-experiments/qiskit_experiments/framework/json.py:581: UserWarning: Could not deserialize instance of class <class 'qiskit_ibm_runtime.ibm_backend.IBMBackend'> from settings {}. Value is {'class': <class 'qiskit_ibm_runtime.ibm_backend.IBMBackend'>, 'settings': {}, 'version': '0.23.0'} 
> The following exception was raised:
> Traceback (most recent call last):
>   File "/tmp/qiskit-experiments/qiskit_experiments/framework/json.py", line 338, in _deserialize_object
>     return cls(**settings)
> TypeError: IBMBackend.__init__() missing 3 required positional arguments: 'configuration', 'service', and 'api_client'
> 
>   return _deserialize_object(obj_val)

The experiment still loads, but this warning is annoying.